Userland/Libraries/LibJS/Runtime/Intl/DurationFormat.cpp

```diff
@@ -141,43 +141,66 @@ ThrowCompletionOr<Temporal::DurationRecord> to_duration_record(VM& vm, Value inp
 
     // 2. Let result be a new Duration Record with each field set to 0.
     Temporal::DurationRecord result = {};
+    bool any_defined = false;
 
-    // 3. Let any be false.
-    auto any = false;
+    auto set_duration_record_value = [&](auto const& name, auto& value_slot) -> ThrowCompletionOr<bool> {
+        auto value = TRY(input_object.get(name));
+        if (value.is_undefined())
+            return false;
 
-    // 4. For each row of Table 1, except the header row, in table order, do
-    for (auto const& duration_instances_component : duration_instances_components) {
-        // a. Let valueSlot be the Value Slot value of the current row.
-        auto value_slot = duration_instances_component.value_slot;
+        value_slot = TRY(Temporal::to_integer_if_integral(vm, value, ErrorType::TemporalInvalidDurationPropertyValueNonIntegral, name, value));
+        return true;
```

Comment:
Alternatively you could set `any_defined = true` here in this case but this works as well

Reply:
Yeah that's much nicer